### ISSUE TITLE
Unable to sign transactions on testnet.

### ISSUE THREAD

**Issue by jhartzell** [blockcypher/node-client#41]
**Please disregard this as the following comments are more accurate**

I have been scouring the internet for a more detailed example on how to actually sign transactions against the BCY testnet but I am not having any luck finding updated information.

Here's the steps I've taken and sorry in advance if I'm doing this completely wrong. I am extremely new to the crypto world.

1. Installed bcwallet to generate a BCY testnet address bprv/bpub key.
2. Created a new transaction from `newTX()` function provided by this library.
3. Attempting to get a ECPair from the bcwallet private key provided.

I attempt to get the ECPair with the following:

```js
const seed = bip32.fromBase58('bprvTxeXPZq5...');
const firstKeyECPair = bitcoin.ECPair.fromPrivateKey(seed.privateKey);
```

This is currently failing with "Invalid network version."

Any help at addressing this issue would be GREATLY appreciated.

**Comment by jhartzell** [comment]
Let me try to further clarify the issues I'm having and the steps I've taken to get here. I feel like I am extremely close to having this functional but am stuck with another error now. (I stopped attempting to use the BCY chain in my example above, although I would like to test on that once I understand these issues)

Steps:

1. Installed electrum wallet and have it functional on the testnet.
2. From the electrum menu I copied the "Master public key"
3. Created an HD wallet via BlockCypher's api with the extended_public_key set to the Master public key I copied in the step above.
4. Derived a new address as demonstrated in the API documentation.
5. Created a transaction via newTX();
6. (This is where I am stuck)

Currently my transaction is being created and sent to me but I am failing to understand how to properly sign the items inside of the `tosign` array.

Here is what I am currently doing (please dont mind the messy code)

```js
let keys: any;
async function getPrivateKey(): Promise<any> {
  if (keys) return keys;

  const seed = await bip39.mnemonicToSeed('<secret stuff here>');
  const root = bitcoin.bip32.fromSeed(seed, bitcoin.networks.testnet);
  const key = bitcoin.ECPair.fromWIF(root.toWIF(), bitcoin.networks.testnet);

  keys = key;

  return keys;
}

async function signTransaction(tx: any) {
  if (!keys) await getPrivateKey();

  tx.pubkeys = [];
  tx.signatures = tx.tosign.map((tosign, n) => {
    tx.pubkeys.push(keys.publicKey.toString('hex'));
    const SIGHASH_ALL = 0x01;
    return bitcoin.script.signature.encode(
      keys.sign(new buffer.Buffer(tosign, 'hex')),
      SIGHASH_ALL,
    ).toString('hex').slice(0, -2);
  });

  console.log(tx);
  console.log(tx.tx.inputs);

  return tx;
}
```

Unless I am completely misunderstanding this then I think my problem is due to me trying to sign with the root node's private key instead of the derived addresses private key?

The error I am getting with this code is: 
```
Error validating generated transaction: Error running script for input 0 referencing ff64d1b732a0f144974ac0e50f7b61f9a88483bdb9a440a85f03b75df3a4a277 at 1: Script was NOT verified successfully.
```

Any information that someone could provide to get me on the right track would be awesome. I feel like I'm so close.

**Comment by jhartzell** [comment]
Alright so upon looking into this further, I was able to find a few bits of information that may be helpful.

When creating a wallet via BlockCypher it asks you to provide an xpub for the HD wallet. I provided the vpub from electrum to this endpoint. When trying to convert the mnemonic seed from the electrum wallet to an ECPair via bitcoinjs-lib I am noticing that the root node is giving me a different xpub than the one I used to create the HD wallet. Could this be the problem?
